pdns/pdns_recursor.cc

```diff
@@ -861,6 +873,42 @@ static bool checkOutgoingProtobufExport(LocalStateHolder<LuaConfigItems>& luacon
 }
 #endif /* HAVE_PROTOBUF */
 
+#ifdef NOD_ENABLED
+static void nodCheckNewDomain(const DNSName& dname)
+{
+  static const QType qt(QType::A);
+  static const uint16_t qc(QClass::IN);
+  // First check the (sub)domain isn't whitelisted for NOD purposes
+  if (!g_nodDomainWL.check(dname)) {
+    // Now check the NODDB (note this is probablistic so can have FNs/FPs)
+    if (t_nodDBp && t_nodDBp->isNewDomain(dname)) {
+      if (g_nodLog) {
+        // This should probably log to a dedicated log file
+        g_log<<Logger::Notice<<"Newly observed domain nod="<<dname.toLogString()<<endl;
+      }
+      if (!(g_nodLookupDomain.compare("") == 0)) {
+        // Send a DNS A query to <domain>.g_nodLookupDomain
+        DNSName qname = dname;
+        vector<DNSRecord> dummy;
+        qname.appendRawLabel(g_nodLookupDomain);
```

Comment:
This will only add one label, so I don't think it that's what you want.

Reply:
suggestions for what you think I might want would be welcome :)

Reply:
Of course, sorry :-) I think the easiest way would be to make `g_nodLookupDomain` a `DNSName`, then `qname += g_nodLookupDomain` should do the needful. You would need to replace `g_nodLookupDomain.compare("") == 0` with `g_nodLookupDomain.empty()`, and I think you're set.

Reply:
Oh I see, so even if g_nodLookupDomain contains "foo.bar.com", it will only append "foo"?

Reply:
Actually no, it would append a single label of length 11, ie `\11foo.bar.com`.

Reply:
Because that's not what I see - I see the whole domain being appended

Reply:
So, if the original name is `www`, do you get `\3www\3foo\3bar\3com\0` or `\3www\11foo.bar.com\0`?

Reply:
I haven't tested to be clear :)

Reply:
Oh I see, I just looked cursorily at the lookup - you're right, it's treating "foo.bar.com" as a single label

Reply:
Ok modified to append the raw labels one at a time